Sources/UtiliKit/ScrollingPageControl/ScrollingPageControl.swift

```diff
@@ -0,0 +1,409 @@
+                self.pageOffset = 0
+                return
+            }
+            
+            guard let pageOffset = pageOffset else { return }
+            
+            // Shift internal stack view
+            stackViewOffsetConstraint.constant = CGFloat(pageOffset) * (dotSize.width + dotSpacing)
+            
+            // Un-scale dots in the middle
+            let mainDotRange = (0..<mainDotCount).map { $0 + marginDotCount + pageOffset }
+            mainDotRange.forEach {
+                guard 0 <= $0 && $0 < dotContainerViews.count else { return }
+                dotContainerViews[$0].transform = .identity
+            }
+            
+            // Scale dots in the margins
+            if let firstMainDotIndex = mainDotRange.first, let lastMainDotIndex = mainDotRange.last {
+                (0..<marginDotCount).forEach {
+                    let scaleFactor = minimumDotScale + (1.0 - minimumDotScale) * (CGFloat($0 + 1) / CGFloat(marginDotCount + 1))
+                    func scaleDot(at index: Int) {
+                        guard index >= 0 && index < dotContainerViews.count else { return }
+                        dotContainerViews[index].transform = needsToScroll ? CGAffineTransform(scaleX: scaleFactor, y: scaleFactor) : .identity
+                    }
+                    let marginOffset = marginDotCount - $0
+                    scaleDot(at: firstMainDotIndex - marginOffset)
+                    scaleDot(at: lastMainDotIndex + marginOffset)
+                }
+            }
+        }
+    }
+    
+    // MARK: Public
+    /// Call to update the appearance of the dot at the specified `index`. Uses `customPageDotAtIndex` to get the new view for `index`.
+    func updateDot(at index: Int) {
+        dotContainerViews[index].subviews.first?.removeFromSuperview()
+        let dotView = customPageDotAtIndex?(index) ?? PageDotView(frame: CGRect(origin: .zero, size: dotSize))
+        dotView.translatesAutoresizingMaskIntoConstraints = false
+        dotContainerViews[index].addSubview(dotView)
+        dotView.centerViewInSuperview()
+    }
+    
+    /// Call to update the appearance of the dot at the specified `indices`. Uses `customPageDotAtIndex` to get the new view for each index in `indices`.
+    func updateDots(at indices: [Int]) {
+        indices.forEach { updateDot(at: $0) }
+    }
+    
+    // MARK: Initalizers
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        initalSubviewSetup()
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        initalSubviewSetup()
+    }
+    
+    convenience init() {
+        self.init(frame: .zero)
+    }
+    
+    // MARK: UIView
+    override var intrinsicContentSize: CGSize {
+        let visibleDots = CGFloat(min(numberOfPages, maxVisibleDots))
+        return CGSize(width: (visibleDots * dotSize.width) + ((visibleDots - 1.0) * dotSpacing), height: max(37.0, dotSize.height))
```

Comment:
Consider extracting the `37.0` into a property so that its purpose can be better understood (min height of `ScrollingPageControl`, right?).